drivers/mtd/nand/raw/atmel_nand.c

```diff
@@ -510,9 +498,26 @@ static int pmecc_correction(struct mtd_info *mtd, u32 pmecc_stat, uint8_t *buf,
 
 			err_nbr = pmecc_err_location(mtd);
 			if (err_nbr == -1) {
-				dev_err(host->dev, "PMECC: Too many errors\n");
-				mtd->ecc_stats.failed++;
-				return -EBADMSG;
+				/* SAMA5D4 PMECC IP can correct errors for all 0xff page */
+				if (host->pmecc_version < PMECC_VERSION_SAMA5D4) {
+					ecc_pos = ecc + i * host->pmecc_bytes_per_sector;
+
+					err_nbr = nand_check_erased_ecc_chunk(
+						buf_pos, host->pmecc_sector_size,
+						ecc_pos, host->pmecc_bytes_per_sector,
+						NULL, 0, (host->pmecc_corr_cap * 3) / 4);
```

Comment:
why the last parameter of nand_check_erased_ecc_chunk() has value (host->pmecc_corr_cap * 3) / 4 ? Shouldn't this be just host->pmecc_corr_cap?

Reply:
Hi @ambarus,

You are right! Just host->pmecc_corr_cap is correct. To be honest I don't remember, where this calculation came from.

Reply:
Hi, @ksdd!

No worries, I've already corrected it. Just FYI, I sent the patch on your behalf to u-boot mainline. Once it gets applied by Eugen, and also backported to our u-boot-at91, we'll close this PR.

Cheers,
ta